beets/autotag/mb.py

```diff
@@ -618,65 +618,49 @@ def album_info(release: Dict) -> beets.autotag.hooks.AlbumInfo:
         )
 
     # We might find links to external sources (Discogs, Bandcamp, ...)
-    if any(
-        config["musicbrainz"]["external_ids"].get().values()
-    ) and release.get("url-relation-list"):
-        discogs_url, bandcamp_url, spotify_url = None, None, None
-        deezer_url, beatport_url, tidal_url = None, None, None
-        fetch_discogs, fetch_bandcamp, fetch_spotify = False, False, False
-        fetch_deezer, fetch_beatport, fetch_tidal = False, False, False
-
-        if config["musicbrainz"]["external_ids"]["discogs"].get():
-            fetch_discogs = True
-        if config["musicbrainz"]["external_ids"]["bandcamp"].get():
-            fetch_bandcamp = True
-        if config["musicbrainz"]["external_ids"]["spotify"].get():
-            fetch_spotify = True
-        if config["musicbrainz"]["external_ids"]["deezer"].get():
-            fetch_deezer = True
-        if config["musicbrainz"]["external_ids"]["beatport"].get():
-            fetch_beatport = True
-        if config["musicbrainz"]["external_ids"]["tidal"].get():
-            fetch_tidal = True
+    external_ids = config["musicbrainz"]["external_ids"].get()
+    wanted_ids = {site for site, wanted in external_ids.items() if wanted}
+    if "url-relation-list" in release and wanted_ids:
+        urls = {}
```

Comment:
You can simplify this entire block as

```python
from itertools import product

...

    external_ids = config["musicbrainz"]["external_ids"].get()
    wanted_sources = {site for site, wanted in external_ids.items() if wanted}
    if wanted_sources and (urls := release.get("url-relation-list", [])):
        urls = {}

        for source, url in product(wanted_sources, urls):
            if f"{source}.com" in (target := url["target"]):
                urls[source] = target
                log.debug(
                    "Found link to {} release via MusicBrainz",
                    source.capitalize(),
                )
```

See the diff below

![image](https://github.com/user-attachments/assets/5d7800f8-99af-4fe7-818a-29714963d824)

Reply:
I made a small tweak before applying it because 2 vars were now called `urls`.

It seems no test actually exercises this path :slightly_frowning_face:  
I did do a real import + checked the result though, so it should be working.